File: mysql_examples/recipes/api/01_connect/connect2.py
#!/usr/bin/python
# connect2.py - connect to the MySQL server, showing how to specify
# a port number or Unix domain socket path explicitly.

import MySQLdb

try:
#@ _FRAG1_
  conn = MySQLdb.connect (db = "cookbook",
                          host = "mysql.example.com",
                          port = 3307,
                          user = "cbuser",
                          passwd = "cbpass")
#@ _FRAG1_
except:
  print "Cannot connect to server"
else:
  print "Connected"
  conn.close ()
  print "Disconnected"

try:
#@ _FRAG2_
  conn = MySQLdb.connect (db = "cookbook",
                          host = "localhost",
                          unix_socket = "/var/tmp/mysql.sock",
                          user = "cbuser",
                          passwd = "cbpass")
#@ _FRAG2_
except:
  print "Cannot connect to server"
else:
  print "Connected"
  conn.close ()
  print "Disconnected"

File: mysql_examples/recipes/api/01_connect/connect2.php
<?php
# connect2.php - connect to the MySQL server, showing how to specify
# the DSN parameters using an array, and how to specify
# a port number or Unix domain socket path explicitly.

require_once "DB.php";

#@ _FRAG0_
$dsn = array
(
  "phptype"  => "mysqli",
  "username" => "cbuser",
  "password" => "cbpass",
  "hostspec" => "localhost",
  "database" => "cookbook"
);
$conn =& DB::connect ($dsn);
if (PEAR::isError ($conn))
  print ("Cannot connect to server\n");
#@ _FRAG0_
else
{
  print ("Connected\n");
  $conn->disconnect ();
  print ("Disconnected\n");
}

#@ _FRAG1_
$dsn = array
(
  "phptype"  => "mysqli",
  "username" => "cbuser",
  "password" => "cbpass",
  "hostspec" => "mysql.example.com",
  "port"     => 3307,
  "database" => "cookbook"
);
$conn =& DB::connect ($dsn);
if (PEAR::isError ($conn))
  print ("Cannot connect to server\n");
#@ _FRAG1_
else
{
  print ("Connected\n");
  $conn->disconnect ();
  print ("Disconnected\n");
}

#@ _FRAG2_
$dsn = array
(
  "phptype"  => "mysqli",
  "username" => "cbuser",
  "password" => "cbpass",
  "hostspec" => "localhost",
  "socket"   => "/var/tmp/mysql.sock",
  "database" => "cookbook"
);
$conn =& DB::connect ($dsn);
if (PEAR::isError ($conn))
  print ("Cannot connect to server\n");
#@ _FRAG2_
else
{
  print ("Connected\n");
  $conn->disconnect ();
  print ("Disconnected\n");
}

?>
